src/smart-components/Channel/context/hooks/useSendVoiceMessageCallback.ts

```diff
@@ -0,0 +1,91 @@
+import { useCallback } from 'react';
+import { GroupChannel } from '@sendbird/chat/groupChannel';
+import { FileMessage, FileMessageCreateParams, MessageMetaArray, UserMessage } from '@sendbird/chat/message';
+import { Logger } from '../../../../lib/SendbirdState';
+import * as messageActionTypes from '../dux/actionTypes';
+import * as utils from '../utils';
+import * as topics from '../../../../lib/pubSub/topics';
+
+interface DynamicParams {
+  currentGroupChannel: GroupChannel;
+  onBeforeSendVoiceMessage?: (file: File, quoteMessage?: UserMessage | FileMessage) => FileMessageCreateParams;
+}
+interface StaticParams {
+  logger: Logger;
+  pubSub: any;
+  scrollRef: React.RefObject<HTMLDivElement>;
+  messagesDispatcher: (props: { type: string, payload: any }) => void;
+}
+type FuncType = (file: File, duration: number, quoteMessage: UserMessage | FileMessage) => void;
+
+export const useSendVoiceMessageCallback = ({
+  currentGroupChannel,
+  onBeforeSendVoiceMessage,
+}: DynamicParams,
+  {
+    logger,
+    pubSub,
+    scrollRef,
+    messagesDispatcher,
+  }: StaticParams): Array<FuncType> => {
+  const sendMessage = useCallback((file: File, duration: number, quoteMessage: UserMessage | FileMessage) => {
+    const messageParams: FileMessageCreateParams = (
+      onBeforeSendVoiceMessage
+      && typeof onBeforeSendVoiceMessage === 'function'
+    )
+      ? onBeforeSendVoiceMessage(file, quoteMessage)
+      : {
+        file,
+        fileName: 'Voice message.mp3',
```

Comment:
`Voice message.mp3` are we really gonna use this name? No timestamp? 
PS. make this const also :D